### DISCUSSION TITLE
Support for encryption/decryption

### DISCUSSION THREAD

**Discussion by MathieuCesbron** [Web3Auth/Web3Auth#523] [Feature Requests]
Hello,
I would like to make a Dapp that needs to encrypt sensitive data on-chain. The data should be encrypted with the receiver public key so only him can decrypt with his private key. This feature is supported by MetaMask for example and it allows the creation of a lot of cool Dapp.

No wallets in Solana are providing this feature as far as I know, I have seen online that I am not the only one missing this feature on Solana. 

Have a good day,

**Comment by itsinz** [comment]
@chaitanyapotti @metallicalfa2

**Comment by writepavel** [comment]
> Hello,
> 
> I would like to make a Dapp that needs to encrypt sensitive data on-chain. The data should be encrypted with the receiver public key so only him can decrypt with his private key. This feature is supported by MetaMask for example and it allows the creation of a lot of cool Dapp.
> 
> 
> 
> No wallets in Solana are providing this feature as far as I know, I have seen online that I am not the only one missing this feature on Solana. 
> 
> 
> 
> Have a good day,

Hi Mathieu, have you found anything? I'm also searching a way to use some Account Abstraction together with user data encryption.

**Reply by MathieuCesbron** [reply]
I have found something and used it in the project: https://github.com/MathieuCesbron/unidoublev2
So basically the browser encrypt something, then give a password to remember for the customer. The customer of the DApp has to remember his password to decrypt the data.

It is annoying to ask the customer to remember a password but that was the only solution at that time.
